src/main/groovy/life/qbic/portal/sampletracking/ui/SampleModifyControls.groovy

```diff
@@ -33,48 +34,49 @@ class SampleModifyControls extends VerticalLayout {
         // Add textfield showing email address of portal user
         userEmailField = new Label()
         userEmailField.setValue("You are not logged in.")
-        /* For Test purposes
-        //TODO: remove in release
-        userEmailField.setValue("CurrentDev@Qbic.com")
-        */
-        this.addComponent(userEmailField)
 
         // Add menu allowing location picking for new sample
         locationSelectMenu = new NativeSelect<>("Select Sample Location")
-
         /* For Test purposes
         //TODO remove in release
         locationSelectMenu.setItems("QBiC", "Home Office", "Laboratory", "Shipping")
         */
-        this.addComponent(locationSelectMenu)
-
 
         // Add menu allowing date picking for new sample
         dateChooser = new DateField("Desired Arrival Date")
         dateChooser.setTextFieldEnabled(false)
         //TODO: choose date format to display
-        this.addComponent(dateChooser)
 
         // Add menu allowing status selection for new sample
         statusSelectMenu = new NativeSelect<Status>("Desired Sample Status")
         statusSelectMenu.setEmptySelectionAllowed(false)
         statusSelectMenu.setItems(Status.values())
+
         /* For Test purposes
         //TODO: remove in release
         sampleSelectMenu.setItems("Waiting", "Processing", "Processed")
         */
-        this.addComponent(statusSelectMenu)
 
         // Add button enabling sample update to SampleList
         updateSampleButton = new Button("Update Samples")
-        this.addComponent(updateSampleButton)
 
         // Add clear Button to delete samples from SampleList
         clearButton = new Button("Clear List")
-        this.addComponent(clearButton)
+
+        // Add all Vaadin components to layout
+        this.addComponents(userEmailField, locationSelectMenu, dateChooser, statusSelectMenu, updateSampleButton, clearButton)
     }
 
     private void registerListeners() {
+
+        // Add listener to update button to upload Sample changes selected in view
+
+        //ToDo Determine how Samples from Samplelist can be connected to user selected location, date and Status
+        def selectedSampleIdArray = ViewModel.requestSampleList().collect { it.getCode }
+
+        //ToDo Date and responsible Persons are stored in Location, but arrivalDate gets selected here, how is this resolved?
+        this.updateSampleButton.addClickListener({ event -> controller.updateSamples(selectedSampleIdArray, locationSelectMenu.getValue(), statusSelectMenu.getValue()) })
```

Comment:
The other values also need binding to the view model. So that here all values can be taken directly from the view model.